### PR TITLE
Convert more transforms to the new API

### PR DESCRIPTION
**Pull request by wolfs**
More dogfooding!

### CODE REVIEW

**Comment by wolfs** [comment]
@gradle/kotlin-dsl: Can you have a look at the converted transforms related to Kotlin? That would be the parameter names transform and the distribution sources transform.

File: buildSrc/subprojects/binary-compatibility/src/main/groovy/org/gradle/binarycompatibility/transforms/FindGradleJar.groovy [hunk 1]
```diff
@@ -17,25 +17,43 @@
...
 
+@AssociatedTransformAction(FindGradleJarAction)
 @CompileStatic
-class FindGradleJar extends ArtifactTransform {
-    private final String target
+interface FindGradleJar {
+    @Input
+    String getTarget()
+    void setTarget(String target)
+}
```
Comment (lptr) [inline buildSrc/subprojects/binary-compatibility/src/main/groovy/org/gradle/binarycompatibility/transforms/FindGradleJar.groovy]: I still find it confusing from the user's point of view that sometimes we use the _verb-like_ name (e.g. `ExplodeZipAndFindJars`) to refer to the action itself, and sometimes (as in the case of `FindGradleJar`) we refer to the parameter object, while the actual action gets an `Action` suffix.

File: buildSrc/subprojects/kotlin-dsl/src/main/kotlin/build/GradleApiParameterNamesTransform.kt [hunk 2]
```diff
@@ -64,11 +67,12 @@ fun Project.withCompileOnlyGradleApiModulesWithParameterNames(vararg gradleModul
     }
 
     dependencies {
-        registerTransform {
+        registerTransform(GradleApiParameterNamesTransform::class) {
```
Comment (lptr) [inline buildSrc/subprojects/kotlin-dsl/src/main/kotlin/build/GradleApiParameterNamesTransform.kt]: We definitely should have a reified version of this.

File: buildSrc/subprojects/kotlin-dsl/src/main/kotlin/build/GradleApiParameterNamesTransform.kt [hunk 3]
```diff
@@ -85,21 +89,37 @@ fun Project.withCompileOnlyGradleApiModulesWithParameterNames(vararg gradleModul
 }
 
 
+@AssociatedTransformAction(GradleApiParameterNamesTransformAction::class)
 internal
-class GradleApiParameterNamesTransform @Inject constructor(
-    private val publicApiIncludes: List<String>,
-    private val publicApiExcludes: List<String>
-) : ArtifactTransform() {
+interface GradleApiParameterNamesTransform {
```
Comment (lptr) [inline buildSrc/subprojects/kotlin-dsl/src/main/kotlin/build/GradleApiParameterNamesTransform.kt]: Is our recommendation to have `Transform` in the name of the transform action/parameter object? I think we should figure out if we recommend using verbs of nouny names with `Transform` as a suffix.

File: buildSrc/subprojects/kotlin-dsl/src/main/kotlin/build/GradleApiParameterNamesTransform.kt (hunk 2 above)
Comment (wolfs) [reply buildSrc/subprojects/kotlin-dsl/src/main/kotlin/build/GradleApiParameterNamesTransform.kt]: I guess we should not for now, see this discussion on [Slack](https://gradle.slack.com/archives/C0WTFHWT1/p1550565252037400?thread_ts=1550564866.036600&cid=C0WTFHWT1):

> Paul Merlin   [2 days ago]
Using the plain Java API would give:
```registerTransform(Something::class.java) { ... }```
which isn’t nice.
> 
> But we also automatically generate `KClass` taking extensions for `Class` taking functions of the Gradle API. Meaning you can already do:
```registerTransform(Something::class) { ... }```
which is good enough.
> 
> Stefan Wolf   [2 days ago]
That is what I thought. So we only add the reified versions for frequently used APIs I guess?
>
> Paul Merlin   [2 days ago]
We also have plans to generate reified extension functions for the Gradle API, uniformly, but we won’t get to it soon.
>
> Paul Merlin   [2 days ago]
Exactly

**Pull request merged**